### PR TITLE
[Fix #5713] `Style/CommentAnnotation` reports only first offending line

### LINKED ISSUE #5713: Style/CommentAnnotation reports only first of multiple offending lines
Good morning everyone,

I stumbled upon some unexpected behavior when working on a project. I created a minimal working example [here](https://gist.github.com/jdno/ba9c04ad7f297a1e2745988d6810b6fc).

I am working on a class that has multiple `# TODO ...` annotations. Something like this:

```ruby
class ToBeDone
  ITEMS = [
    '', # TODO Item 1
    '', # TODO Item 2

    '', # TODO Item 3
  ].freeze
end
```

When running Rubocop, it reports a `Style/CommentAnnotation` offense, because the proper style would be `# TODO: ...`. But it does that only for the first and third occurrence, and not for the second one.

```
jandavid@STS ~/D/j/rubocop-todo-bug> rubocop
Inspecting 2 files
C.

Offenses:

todo.rb:5:11: C: Style/CommentAnnotation: Annotation keywords like TODO should be all upper case, followed by a colon, and a space, then a note describing the problem.
    '', # TODO Item 1
          ^^^^^
todo.rb:8:11: C: Style/CommentAnnotation: Annotation keywords like TODO should be all upper case, followed by a colon, and a space, then a note describing the problem.
    '', # TODO Item 3
          ^^^^^

2 files inspected, 2 offenses detected
```

Even after fixing the first and third line, the second line is never reported as an offense. This also means that `rubicon -a` will never pick that line up and fix it automatically.

## Expected behavior

When there are `Style/CommentAnnotation` in multiple lines that follow each other, Rubocop should report a violation for each individual line.

## Actual behavior

When there are `Style/CommentAnnotation` in multiple lines that follow each other, only the first offense is reported.

## Steps to reproduce the problem

Check out the [minimal working example](https://gist.github.com/jdno/ba9c04ad7f297a1e2745988d6810b6fc). Download the Gist as a ZIP, extract the archive, execute `bundle` if necessary and run `rubocop` in it.

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
jandavid@STS ~/D/j/rubocop-todo-bug> rubocop -V
0.54.0 (using Parser 2.5.0.4, running on ruby 2.5.0 x86_64-darwin17)
```

### PR DESCRIPTION
**Pull request by svendittmer**
The cop allows keywords in mutli-line sentences, so it checks only the first line for an offense.
This makes sense if a line consists only of a comment. If comments are appended to code however, they are not part of a multi-line sentence, so each of them should be checked for offenses. 

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: spec/rubocop/cop/style/comment_annotation_spec.rb [hunk 1]
```diff
@@ -113,12 +113,27 @@
...
+  context 'comments in consecutive lines' do
+    it 'accepts a keyword that is somewhere in a sentence' do
+      expect_no_offenses(<<-RUBY.strip_indent)
+        # Example: There are three reviews, with ranks 1, 2, and 3. A new
+        # review is saved with rank 2. The two reviews that originally had
+        # ranks 2 and 3 will have their ranks increased to 3 and 4.
+      RUBY
+    end
+
+    it 'adds an offense for each appended comment' do
```
Comment (bbatsov) [inline spec/rubocop/cop/style/comment_annotation_spec.rb]: I wonder if appended comment is the best way to name this. To me it sounds confusing.

File: spec/rubocop/cop/style/comment_annotation_spec.rb [hunk 2]
```diff
@@ -113,12 +113,27 @@
     RUBY
   end
 
-  it 'accepts a keyword that is somewhere in a sentence' do
-    expect_no_offenses(<<-RUBY.strip_indent)
-      # Example: There are three reviews, with ranks 1, 2, and 3. A new
-      # review is saved with rank 2. The two reviews that originally had
-      # ranks 2 and 3 will have their ranks increased to 3 and 4.
-    RUBY
+  context 'comments in consecutive lines' do
```
Comment (bbatsov) [inline spec/rubocop/cop/style/comment_annotation_spec.rb]: I'm not certain that we should bucket a "multi-line" comment and consecutive inline comments together.

File: spec/rubocop/cop/style/comment_annotation_spec.rb (hunk 1 above)
Comment (svendittmer) [reply spec/rubocop/cop/style/comment_annotation_spec.rb]: How about trailing comment?

File: spec/rubocop/cop/style/comment_annotation_spec.rb (hunk 2 above)
Comment (svendittmer) [reply spec/rubocop/cop/style/comment_annotation_spec.rb]: I regard multi-line comments containing keywords as an exception from the rule, and consecutive inline comments as exception from the exception. That's also how I implemented the fix. My intention bucketing them together was to show this relationship, but I wasn't certain, too.

File: spec/rubocop/cop/style/comment_annotation_spec.rb (hunk 1 above)
Comment (bbatsov) [reply spec/rubocop/cop/style/comment_annotation_spec.rb]: I think in most of our code we call those "inline comments". You can grep the code to check.

File: spec/rubocop/cop/style/comment_annotation_spec.rb (hunk 2 above)
Comment (bbatsov) [reply spec/rubocop/cop/style/comment_annotation_spec.rb]: I'd rather just slit the context to avoid the potential confusion.

**Pull request merged**